concepts/&str/about.md

```diff
@@ -1,3 +1,69 @@
 # About
 
-TODO
+There are two ways that Rust implements an array of characters: [str][str] and [String][String].
+
+`str`, also known as a "string slice", is usually used with a reference as `&str`.
+
+To know why, it may help to understand why the following line of code will not compile
+
+```rust
+const my_const: str = "Hello, World!";
+
+// the size for values of type `str` cannot be known at compilation time
+// the trait `Sized` is not implemented for `str`
+```
+
+Even though the literal "Hello, World!" length can be known at compilation time, as a `str` it is a sequence of unknown size.
+But as a _reference_ to a literal (`&str`), it has a "fat pointer" which holds the address and length of the literal in memory.
+Fat pointers are of a known size, and so this line will compile
+
+```rust
+const my_const: &str = "Hello, World!";
+```
+
+A mutable string slice (`mut &str`) can have its valued replaced, but not modified in place.
+
+Example
+
+```rust
+let mut my_str = "Hello, World!";
+my_str = "Hi!";
+let mut my_string: String = String::from("Hello");
+my_string.push_str(", World!"); // there is no equivalent function to modify &str
+```
+
+It is less constraining to use `&str` instead of `&String` when defining a parameter for a function, as this will not compile
+
+```rust
+pub fn main() {
+    let my_str: &str = "Hello, World!";
+    say_hi(my_str);
+}
+
+fn say_hi(phrase: &String) {
+    println!("{:?}", phrase);
+}
+
+// mismatched types
+// expected reference `&String`
+// found reference `&str`
+```
+
+A `&str` will not be [coerced][coercion] into a `&String`. But this will compile
+
+```rust
+pub fn main() {
+    let my_string: String = String::from("Hello, World!");
```

Comment:
```suggestion
    let my_string = String::from("Hello, World!");
```

An explicit type there is redundant.

Reply:
Yeah, I had it explicit for pedagogic purposes, but I think you're right to keep it best practice.